consensus/src/consensus/inner.rs

```diff
@@ -0,0 +1,508 @@
+                            self.decommit_ledger_block(&fork_path.path[0]).await?;
+
+                            for block_hash in fork_path.path {
+                                if &block_hash == hash {
+                                    self.process_block(&hash, block).await?
+                                } else {
+                                    let new_block = self.storage.get_block(&block_hash).await?;
+                                    self.process_block(&new_block.header.hash(), &new_block).await?;
+                                }
+                            }
+                        } else {
+                            // dont process
+                            return Ok(());
+                        }
+                    }
+                    ForkDescription::Orphan => {
+                        debug!("Processing a block that is on unknown orphan chain");
+                        // dont process
+                        return Ok(());
+                    }
+                }
+            }
+        }
+        self.process_block(&hash, block).await?;
+        Ok(())
+    }
+
+    async fn commit_block(&mut self, hash: &Digest, block: &SerialBlock) -> Result<(), ConsensusError> {
+        let mut commitments = vec![];
+        let mut serials = vec![];
+        let mut memos = vec![];
+        for transaction in block.transactions.iter() {
+            commitments.extend(transaction.new_records.iter().map(|x| x.commitment.clone()));
+            serials.extend(transaction.old_serial_numbers.iter().cloned());
+            memos.push(transaction.memorandum.clone());
+        }
+        let digest = self.ledger.extend(&commitments[..], &serials[..], &memos[..])?;
+        self.storage.commit_block(hash, digest).await?;
+        Ok(())
+    }
+
+    /// removes a block and all of it's descendents from storage and ledger
+    async fn decommit_ledger_block(&mut self, hash: &Digest) -> Result<(), ConsensusError> {
+        let decommited_blocks = self.storage.decommit_blocks(hash).await?;
+        let mut commitments = vec![];
+        let mut serials = vec![];
+        let mut memos = vec![];
+        for block in decommited_blocks.into_iter() {
+            for transaction in block.transactions.iter() {
+                commitments.extend(transaction.new_records.iter().map(|x| x.commitment.clone()));
+                serials.extend(transaction.old_serial_numbers.iter().cloned());
+                memos.push(transaction.memorandum.clone());
+            }
+        }
+        self.ledger.rollback(&commitments[..], &serials[..], &memos[..])?;
+        Ok(())
+    }
+
+    /// Return whether or not the given block is valid and insert it.
+    /// 1. Verify that the block header is valid.
+    /// 2. Verify that the transactions are valid.
+    /// 3. Insert/canonize block.
+    async fn process_block(&mut self, hash: &Digest, block: &SerialBlock) -> Result<(), ConsensusError> {
```

Comment:
This is called by `try_commit_block`, so I think this name should be changed to something that reflects the fact that it's not processing just any block, but one that is about to be committed.